File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch04/Python/P04-Sort2.py
# Ascending indexing of two correlated arrays
from sort import *

# main

n = 6                                         # number of values to be sorted
x = [0., 40., 60., 40., 20., 60., 20.]                        # primary array
y = [0.,  2.,  2.,  1.,  2.,  1.,  1.]                     # secondaryy array
ind = [0]*(n+1)                                            # array of indexes

print("Original arrays:")
print("x",end="")
for i in range(1,n+1): print("{0:6.2f}".format(x[i]),end="")
print()
print("y",end="")
for i in range(1,n+1): print("{0:6.2f}".format(y[i]),end="")
print()

Index2(x,y,ind,n)
print("Indexes:")
for i in range(1,n+1): print("{0:6d}".format(ind[i]),end="")
print()

print("Indexed arrays:")
print("x",end="")
for i in range(1,n+1): print("{0:6.2f}".format(x[ind[i]]),end="")
print()
print("y",end="")
for i in range(1,n+1): print("{0:6.2f}".format(y[ind[i]]),end="")
print()

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch04/C/P04-Sort2.cpp
// Ascending indexing of two correlated arrays
#include <stdio.h>
#include "sort.h"

int main()
{
   const int n = 6;                          // number of values to be sorted
   double x[] = {0., 40., 60., 40., 20., 60., 20.};          // primary array
   double y[] = {0.,  2.,  2.,  1.,  2.,  1.,  1.};        // secondary array
   int ind[n+1];                                          // array of indexes
   int i;

   printf("Original arrays:\n");
   printf("x");
   for (i=1; i<=n; i++) printf("%6.2f",x[i]); printf("\n");
   printf("y");
   for (i=1; i<=n; i++) printf("%6.2f",y[i]); printf("\n");

   Index2(x,y,ind,n);
   printf("Indexes:\n");
   for (i=1; i<=n; i++) printf("%6i",ind[i]); printf("\n");

   printf("Indexed arrays:\n");
   printf("x");
   for (i=1; i<=n; i++) printf("%6.2f",x[ind[i]]); printf("\n");
   printf("y");
   for (i=1; i<=n; i++) printf("%6.2f",y[ind[i]]); printf("\n");
}
